app/views.py

```diff
@@ -55,30 +57,43 @@ def results(request):
             data_to_render['error_message'] = 'Sorry, we couldn\'t find a move with that title.'
             return render(request, 'index.html', data_to_render)
 
-        # get list of Statuses about the movie
-        raw_tweets = []
-        try:
-            raw_tweets = TwitterAPI().search_movie(movie)
-        except Exception as error:
-            if type(error) is ValueError:
-                print('ERROR: Rate limit exceeded')
-                data_to_render['error_message'] = 'Sorry, SilverScreen\'s Twitter API rate limit has been exceeded. Please try a different movie, or try again later.'
+        # attempt to fetch tweets from the database
+        clean_tweets = Tweet.objects.filter(imdbID = movie.imdbID)
+
+        # if we have no tweets about the movie
+        if not clean_tweets:
```

Comment:
Weighing newer tweets over older tweets will be handled by the overall score generation algorithm (right now we're only assigning scores to individual tweets). We do, however, need to eventually purge old tweets from the database.